src/fmpz_mpoly/mul_array.c

```diff
@@ -15,6 +15,12 @@
 #include "mpoly.h"
 #include "fmpz_mpoly.h"
 
+/* Currently we do not -funroll-loops by default in the fmpz_mpoly module,
+   but it is worthwhile for the basic loops here. */
+#pragma GCC push_options
+#pragma GCC optimize("-funroll-loops")
```

Comment:
I think we either should unroll the loops through options or manually, but not both.

Reply:
There are loops that make sense to unroll manually, others I'd just have the compiler do it for me with an annotation.